crates/forge/bin/cmd/inspect.rs

```diff
@@ -592,6 +595,40 @@ fn print_eof(bytecode: Option<CompactBytecode>) -> Result<()> {
     Ok(())
 }
 
+fn check_revive_field(field: &ContractArtifactField) -> Result<bool> {
+    let fields_revive_specific_behavior =
+        [ContractArtifactField::Bytecode, ContractArtifactField::DeployedBytecode];
+
+    let fields_revive_unimplemented_warn = [
+        ContractArtifactField::GasEstimates,
+        ContractArtifactField::StorageLayout,
+        ContractArtifactField::Metadata,
+        ContractArtifactField::Eof,
+        ContractArtifactField::EofInit,
+    ];
+
+    let fields_revive_should_error = [
+        ContractArtifactField::Assembly,
```

Comment:
dq: but why we dont support `assembly`, `ir` and `ir_optimized`?

Reply:
I assume `assembly` is EVM assembly - not PVM.
I am not sure about `ir`, `ir_optimized` maybe @xermicus could clarify this.

Reply:
if we go by the flow inside `revive-solidity` we set `assembly_text` to PVM assembly [click me](https://github.com/paritytech/revive/blob/66f9a4d64f36777e6a632736121c1cc57d33de16/crates/solidity/src/build/contract.rs#L141). 

`ir` and `ir_optimized` are just `yul`, no?

Reply:
ir and ir_optimizes should be YUL I guess

Reply:
https://docs.soliditylang.org/en/latest/using-the-compiler.html#output-description